soc: stm32: Fix M4 core not booting

cameled opened the pull request:
LL_HSEM_1StepLock() may failed, use z_stm32_hsem_lock() instead.

galak reviewed (changes requested):
Can you update the commit message.

The first line should be something like `soc: stm32: Fix M4 core not booting`

cameled commented:
> Can you update the commit message.
> 
> The first line should be something like `soc: stm32: Fix M4 core not booting`

Done.

Commit:
soc: stm32: Fix M4 core not booting

For stm32h7x dual core SOC, M4 wake-up event triggered by
HSEM1 interrupt. But LL_HSEM_1StepLock() may failed, use
z_stm32_hsem_lock() instead.

Signed-off-by: HaiLong Yang <hailong.yang@brainco.cn>

cameled commented:
Since I'm still not sure why this happened, this pull request is not ready to merge.

cameled commented:
This pull request is not needed.
